src/private/strarray.cpp

```diff
@@ -22,30 +22,48 @@
 namespace LibQGit2 {
 namespace internal {
 
-StrArray::StrArray()
-{
-    m_data.count = 0;
-    m_data.strings = NULL;
-}
+static_assert(!std::is_copy_constructible<StrArray>::value, "StrArray must NOT be copy constructible");
+static_assert(std::is_move_constructible<StrArray>::value, "StrArray must be move constructible");
+static_assert(!std::is_copy_assignable<StrArray>::value, "StrArray must NOT be copy assignable");
+static_assert(std::is_move_assignable<StrArray>::value, "StrArray must be move assignable");
 
 StrArray::StrArray(const QList<QByteArray> &list) :
     m_strings(list)
 {
-    m_data.count = 0;
-    m_data.strings = NULL;
+    m_data.count = size_t(m_strings.size());
+
+    if (m_data.count == 0) {
+        return;
+    }
 
-    updateNative();
+    m_data.strings = static_cast<char**>(malloc(m_data.count * sizeof(char*)));
```

Comment:
Using `malloc/free` isn't ideal for C++ code. I'd prefer storing the character data in a `std::vector<char*>` instance and initializing `m_data.strings` using `vector`'s `data()` method, or at least using `new char*[]/delete[]`.

Reply:
Using `std::vector<char*>` doesn't feel like it would simplify things; actually the opposite. The C strings' (`char` arrays') memory still needs to be handled somehow, i.e. freed at `StrArray` destructor. The `QList<QByteArray> m_strings` takes care of that now. Adding a `std::vector<char*>` would add another member to the class, which would get rid of the `free/delete[]` in the destructor but add additional handling in other methods so that the new member stays in sync with `m_strings`.

I can replace `malloc/free` with `new[]/delete[]`.